scipy/stats/_levy_stable/__init__.py

```diff
@@ -225,6 +230,7 @@ def _pdf_single_value_piecewise_post_rounding_Z0(x0, alpha, beta, quad_eps):
     c2 = _nolan.c2
     g = _nolan.g
 
+    x0 = _nolan_round_x_near_zeta(x0, alpha, zeta, x_tol_near_zeta)
```

Comment:
Yes, the rounding has already occurred, however the value of zeta changes later (line 228). So, the condition at line 233 `if x0 == zeta:` becomes false. This leads to incorrect answers that surprised me when I saw the negative values of PDF at the points where I expected to see the peak. So we really need to do rounding here. Maybe, we could eliminate the first rounding at `_pdf_single_value_piecewise_Z0`, but doing this caused fails of some tests for me.

Reply:
Oh, right. sorry about that. I'll need to take a closer look. It's strange that your getting these failures on such a common platform. I'd like to be able to reproduce this locally.

```I tested on x86_64, Linux 5.4.0-132-generic, Ubuntu 20.04.5 LTS, Python 3.10.10.```

Reply:
In Python zeta is calculated like this

```python
zeta = -beta * np.tan(np.pi * alpha / 2.0)
````

and in C like this

```C
 sp->zeta = -beta * tan(M_PI_2 * alpha);
```

Since the formula's aren't quite the same, C it's not surprising that these values could be different on some platforms. Your right, we shouldn't be assuming that the value computed in C will be the same as that computed in Python

Reply:
Your solution works, I wonder if there's a clearer way to do it though. If we go with yours, we'll need a comment explaining what's happening

Reply:
As an alternative I see
1) Using the old calculated zeta
2) Comparing "x close to zeta" instead of "x == zeta"

Unfortunately, I am not experienced enough to predict do these solutions lead to unexpected behaviour in other places of the algorithm or not.

Reply:
> What C compiler are you using?

Cython 0.29.33
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

To reproduce the error, I advise you to reduce `stats.levy_stable.quad_eps`, for example
```python
stats.levy_stable.quad_eps = 1.2e-11
```
Also, you can use this snippet to search for bad parameters:
```python
while True:
    alpha = np.random.uniform(0, 2, size=2000)
    beta = np.random.uniform(-1, 1, size=2000)
    logpdf = levy_stable.logpdf(0, alpha=alpha, beta=beta)
    if pd.isna(logpdf).any():
        bad = pd.DataFrame(
            dict(
                alpha=alpha[pd.isna(logpdf)],
                beta=beta[pd.isna(logpdf)]
            )
        )
        print('Found! ', bad)
        break
```